README.md

```diff
@@ -13,128 +13,339 @@
 
 ## Supported Languages and versions
-- [Language versions supported in each runtime version](https://docs.microsoft.com/en-us/azure/azure-functions/supported-languages#languages-by-runtime-version)
-- [Languages supported in each OS](https://docs.microsoft.com/en-us/azure/azure-functions/supported-languages#language-support-details)
 
-## Using Publish Profile as Deployment Credential (recommended)
-Using publish profile as deployment credential is recommended if you are not the owner of your Azure subscription. Follow these steps to configure your workflow to use the publish profile from your function app.
++ [Language versions supported in each runtime version](https://docs.microsoft.com/en-us/azure/azure-functions/supported-languages#languages-by-runtime-version)
++ [Languages supported in each OS](https://docs.microsoft.com/en-us/azure/azure-functions/supported-languages#language-support-details)
+
+## Authentication methods
+
+You must choose how the action authenticates with Azure when deploying code from GitHub to your function app. There are three supported authentication methods:
+
+| Method| Description |
+| ---- | ---- |
+| [OpenID Connect (OIDC)](https://docs.github.com/actions/security-for-github-actions/security-hardening-your-deployments/configuring-openid-connect-in-cloud-providers) | This is the recommended and most secure authentication method from GitHub. Supports role-based access control (RBAC) for accessing Azure resources. Leverages user-assigned managed identities, which means that secrets are managed by Azure. |
+| Azure service principal | GitHub uses service principal credentials to access Azure resources using RBAC. You must manage the service principal secrets in GitHub. |  
+| Publish profile | Credentials (username and password) are stored in GitHub and used to access the `scm` endpoint using HTTP basic authentication during deployment. Might require you to enable the `scm` endpoint in your app. |
+
+These are special considerations for certain hosting scenarios:
+
++ When your app is hosted in a sovereign cloud or on Azure Stack Hub, make sure to include the relevant `environment` and `audience` parameters for the `azure/login` action when using OIDC or RBAC authorization.
++ RBAC is the only supported authentication method when your app is hosted on Linux in a Consumption plan and the project contains an executable file, such as a custom handler, or `chrome` in [Puppeteer](https://github.com/puppeteer/puppeteer)/[Playwright](https://github.com/microsoft/playwright).
```

Comment:
This is confusing. RBAC is not a defined one of the methods, and in theory it could refer to two of the options that were defined. I think you're speaking to service state rather than GHA state, but that change isn't clear. The intersection of conditions also reads confusingly. Could we split out sub-bullets for scenarios where basic publishing credentials are not supported?

Reply:
Ah good catch. RBAC is what we were saying before. Now, it should read, "...when authenticating with OIDC or an Azure service principal".

What I'm trying to convey is, "if you want to use OIDC or a service principal don't forget the extra parameters you'll need for `azure/login` if you're in these clouds."